src/platforms/macos/daemon/macosroutemonitor.cpp

```diff
@@ -105,7 +105,7 @@ void MacosRouteMonitor::handleRtmUpdate(const struct rt_msghdr* rtm,
 
   // We expect all useful routes to contain a destination, netmask and gateway.
   if (!(rtm->rtm_addrs & RTA_DST) || !(rtm->rtm_addrs & RTA_GATEWAY) ||
-      !(rtm->rtm_addrs & RTA_NETMASK)) {
+      !(rtm->rtm_addrs & RTA_NETMASK) || (addrlist.count() < 3)) {
```

Comment:
why?

Reply:
For reasons that I don't quite understand, I have seen a few occurrences of routing messages from the kernel which set more flags in `rtm_addrs` than were included as attributes. This would seem to be a violation of the routing socket protocol and it leads to an assert later on when indexing into the `addrlist`. The extra check here ensures that we simply ignore those messages.